sci-mathematics/polymake/polymake-4.2.ebuild

```diff
@@ -0,0 +1,87 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit eutils flag-o-matic
+
+DESCRIPTION="research tool for polyhedral geometry and combinatorics"
+SRC_URI="https://polymake.org/lib/exe/fetch.php/download/${P}.tar.bz2"
+HOMEPAGE="https://polymake.org"
+
+IUSE="+cdd lrs ppl bliss +libnormaliz singular libpolymake"
+
+SLOT="0"
+LICENSE="GPL-2"
+KEYWORDS="~amd64 ~x86 ~amd64-linux ~x86-linux"
+
+DEPEND="dev-lang/perl
+	dev-libs/boost:=
+	dev-libs/gmp:0=
+	dev-libs/mpfr:0
+	dev-libs/libxml2:2
+	dev-libs/libxslt
+	dev-perl/JSON
+	sci-mathematics/flint
+	ppl? ( dev-libs/ppl )
+	cdd? ( sci-libs/cddlib )
+	lrs? ( >=sci-libs/lrslib-051[gmp] )
+	bliss? ( sci-libs/bliss[gmp] )
+	libnormaliz? ( dev-libs/boost:= )
+	singular? ( >=sci-mathematics/singular-4.0.1 )"
+RDEPEND="${DEPEND}
+	dev-perl/XML-LibXML
+	dev-perl/XML-LibXSLT
+	dev-perl/XML-Writer
+	dev-perl/Term-ReadLine-Gnu"
+
+pkg_pretend() {
+	einfo "During compile this package uses up to"
+	einfo "750MB of RAM per process. Use MAKEOPTS=\"-j1\" if"
+	einfo "you run into trouble."
+}
+
+src_configure() {
+	export CXXOPT=$(get-flag -O)
+
+	# We need to define BLISS_USE_GMP if bliss was built with gmp support.
+	# Therefore we require gmp support on bliss, so that the package
+	# manager can prevent rebuilds with changed gmp flag.
+	if use bliss ; then
+		append-cxxflags -DBLISS_USE_GMP
+	fi
```

Comment:
`use bliss && append-cxxflags -DBLISS_USE_GMP`
even better: try if `append-cppflags` works (which is the correct injecter)